src/redux/authenticationSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  isLoading: true,
+  isAuthenticated: false,
+  authToken: '',
+  response: {},
+  user_id: ''
+};
+
+const LOGIN_URL = 'https://beta-yacht-rental.onrender.com/login';
+const LOGOUT_URL = 'https://beta-yacht-rental.onrender.com/logout';
+const SIGNUP_URL = 'https://beta-yacht-rental.onrender.com/signup';
+const CURRENTUSER_URL = 'https://beta-yacht-rental.onrender.com/current_user';
```

Comment:
- [ ] The frontend app is now set to completely consume the server that is on the cloud in all Redux slices, however, it won't work:

![image](https://github.com/HtetWaiYan7191/yacht_rental_front_end/assets/57421823/dca68202-c908-44d3-9485-d4f4c7ae9724)

## Explanation

This is because the `cors.rb` file that has been uploaded to the cloud server needs to be configured to also accept `http://localhost:5173/` as an origin.

Since we reviewers cannot view or access the live demo code, we are required to test the code on our local machines.  After that, if a live demo exists, we also review it.

Also, if the final evaluator and/or a potential employer wants to clone your code to put is to test, they will be greeted with the same issue.

## Possible solutions

### Solution 1. 

On Rails, edit your cors.rb file to accept both the `http://localhost:5173/` and the cloud frontend whose URL seems to be `https://beta-yacht-rental.netlify.app/`.  After that, do a deployment to the cloud so the new `cors.rb` gets updated on the live demo.  

You could specify both origins in the file as explained previously, or just use splat '*' to allow any origin.

_After this, both the local and cloud frontend will be able to consume the cloud API._

### Solution 2 (tested). 

Leave your `cors.rb`, your instructions, and your live demo untouched as they exist now, but change all your Redux slices URLs to consume `http://127.0.0.1:3000`.

This is easily done with a search for `https://beta-yacht-rental.onrender.com` and replace for `http://127.0.0.1:3000`:

![image](https://github.com/HtetWaiYan7191/yacht_rental_front_end/assets/57421823/18f310bb-c74f-4a29-aa27-1b3057bd3765)

_After this, both your server and your frontend will work locally, and the live demo will still work independently._


### I applied solution 2 myself exactly as explained and I was able to successfully get into the app, the console, and into the database:

![image](https://github.com/HtetWaiYan7191/yacht_rental_front_end/assets/57421823/ba19e51c-df1b-40cb-84bf-112d954074a5)

![image](https://github.com/HtetWaiYan7191/yacht_rental_front_end/assets/57421823/ae0ec9c6-4cee-48e2-8cc8-90d6fda1150e)

The app worked well.

Kindly, choose either solution.  That way, your raw code will be able to be tested. 👍